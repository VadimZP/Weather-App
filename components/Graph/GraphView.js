import React from 'react'
import PropTypes from 'prop-types'

import { Group } from '@vx/group'
import { LinePath } from '@vx/shape'
import { curveMonotoneX } from '@vx/curve'
import { LinearGradient } from '@vx/gradient'

GraphView.propTypes = {
    data: PropTypes.arrayOf(PropTypes.object),
    width: PropTypes.number,
    height: PropTypes.number,
    margin: PropTypes.object,
    xScale: PropTypes.func,
    yScale: PropTypes.func,
    x: PropTypes.func,
    y: PropTypes.func,
    glyphDot: PropTypes.func,
}

export default function GraphView(props) {
    return (
        <svg style={{ borderBottom: '1px solid #000' }} width={props.width} height={props.height}>
            <LinearGradient id="gradient" to="#EFEEFC" from="#CBC8EF" />
            <rect
                x={0}
                y={0}
                width={props.width}
                height={props.height}
                stroke="#f1f1f1"
                fill="url(#gradient)"
            />
            <Group top={props.margin.top}>

                <LinePath
                    data={props.data}
                    xScale={props.xScale}
                    yScale={props.yScale}
                    x={props.x}
                    y={props.y}
                    stroke="#7e20dc"
                    strokeWidth={2}
                    curve={curveMonotoneX}
                    glyph={(d, i) => (props.glyphDot(d, i))}
                />
            </Group>
        </svg>
    )
}
