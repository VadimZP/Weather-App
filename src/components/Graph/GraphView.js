import React from 'react'
import PropTypes from 'prop-types'

import { Line } from 'react-chartjs-2'

GraphView.propTypes = {
    chartData: PropTypes.object,
    chartOptions: PropTypes.object,
    getElem: PropTypes.func,
}

export default function GraphView({ chartData, chartOptions, getElem }) {
    if (chartData === undefined) return ''

    return (
        <div className="weather-graph">
            <Line
                data={chartData}
                options={chartOptions}
                ref={el => getElem(el)}
            />
        </div>
    )
}

