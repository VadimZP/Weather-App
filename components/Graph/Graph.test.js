import React from 'react'
import renderer from 'react-test-renderer'

import GraphView from './GraphView'

test('Return GraphView', () => {
    const component = renderer.create(<GraphView
        data={[{}, {}]}
        width={1200}
        height={400}
        margin={{}}
        xScale={() => {}}
        yScale={() => {}}
        x={() => {}}
        y={() => {}}
        glyphDot={() => {}}
    />)
    const GraphViewTree = component.toJSON()
    expect(GraphViewTree).toMatchSnapshot()
})
