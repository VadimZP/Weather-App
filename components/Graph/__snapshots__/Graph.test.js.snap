// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return GraphView 1`] = `
<svg
  height={400}
  style={
    Object {
      "borderBottom": "1px solid #000",
    }
  }
  width={1200}
>
  <defs>
    <linearGradient
      gradientTransform={undefined}
      id="gradient"
      x1="0"
      x2="0"
      y1="0"
      y2="1"
    >
      <stop
        offset="0%"
        stopColor="#CBC8EF"
        stopOpacity={1}
      />
      <stop
        offset="100%"
        stopColor="#EFEEFC"
        stopOpacity={1}
      />
    </linearGradient>
  </defs>
  <rect
    fill="url(#gradient)"
    height={400}
    stroke="#f1f1f1"
    width={1200}
    x={0}
    y={0}
  />
  <g
    className="cx-group"
    transform="translate(0, 0)"
  >
    <g>
      <path
        className="vx-linepath"
        d="MNaN,NaNLNaN,NaN"
        fill="none"
        stroke="#7e20dc"
        strokeDasharray=""
        strokeDashoffset={0}
        strokeWidth={2}
      />
      <g
        className="vx-linepath-glyphs"
      />
    </g>
  </g>
</svg>
`;
